File: core/src/main/java/vn/loitp/function/epub/core/PageFragment.java
package vn.loitp.function.epub.core;

import android.content.Context;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

import loitp.core.R;

/**
 * Created by loitp on 08.09.2016.
 */
public class PageFragment extends Fragment {
    /**
     * The fragment argument representing the section number for this
     * fragment.
     */
    private static final String ARG_TAB_POSITON = "tab_position";

    private OnFragmentReadyListener onFragmentReadyListener;

    /**
     * Returns a new instance of this fragment for the given section
     * number.
     */
    public static PageFragment newInstance(int tabPosition) {
        PageFragment fragment = new PageFragment();
        Bundle args = new Bundle();
        args.putInt(ARG_TAB_POSITON, tabPosition);
        fragment.setArguments(args);
        return fragment;
    }

    public PageFragment() {
    }

    public interface OnFragmentReadyListener {
        View onFragmentReady(int position);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        onFragmentReadyListener = (OnFragmentReadyListener) context;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        onFragmentReadyListener = null;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.frm_epub_reader, container, false);
        RelativeLayout mainLayout = (RelativeLayout) rootView.findViewById(R.id.fragment_main_layout);
        View view = onFragmentReadyListener.onFragmentReady(getArguments().getInt(ARG_TAB_POSITON));
        if (view != null) {
            mainLayout.addView(view);
        }
        return rootView;
    }
}

File: core/src/main/java/com/function/epub/core/PageFragment.kt
package com.function.epub.core

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import androidx.fragment.app.Fragment
import com.R

class PageFragment : Fragment() {

    companion object {
        private const val ARG_TAB_POSITION = "ARG_TAB_POSITION"

        fun newInstance(tabPosition: Int): PageFragment {
            val fragment = PageFragment()
            val bundle = Bundle()
            bundle.putInt(ARG_TAB_POSITION, tabPosition)
            fragment.arguments = bundle
            return fragment
        }
    }

    private var onFragmentReadyListener: OnFragmentReadyListener? = null

    interface OnFragmentReadyListener {
        fun onFragmentReady(position: Int): View?
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        onFragmentReadyListener = context as OnFragmentReadyListener
    }

    override fun onDestroy() {
        super.onDestroy()
        onFragmentReadyListener = null
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.l_frm_epub_reader, container, false)

        arguments?.let {
            val position = it.getInt(ARG_TAB_POSITION)
            val view = onFragmentReadyListener?.onFragmentReady(position)
            view?.let {
                val fragmentMainLayout = rootView.findViewById<RelativeLayout>(R.id.fragmentMainLayout)
                fragmentMainLayout.addView(view)
            }
        }
        return rootView
    }
}
